nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+
+
+'''Method Interface'''
+
+<pre>
+function signMessage({message: string}): Promise<SignedMessage>
+</pre>
+
+
+'''Input arguments'''
+
+<pre>
+interface SignMessageArgs {
+  message: string; // Arbitrary message to sign
+}
+</pre>
+
+
+'''Success return value'''
+
+<pre>
+interface SignedMessage {
+  publicKey: string; // Public key of account that signed message
+  message: string; // Original message signed
+  salt: string; // Salt added to original message as prefix, before signing
+  data; string; // Signed message
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string;
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+signMessage({
+  message: 'Hello World!',
+})
+.then((signedMessage: SignedMessage) => {
+  const {
+    publicKey,
+    message,
+    salt,
+    data,
+  } = signedMessage;
+
+  console.log('Public key used to sign:', publicKey);
+  console.log('Original message:', message);
+  console.log('Salt added to message:', salt);
+  console.log('Signed data:', data);
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case UNKNOWN_ERROR:
+      console.log(description);
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request
+* Present the message to the user for signing
+* Create a randomized salt value to be prefixed to the message before signing
+* Create the hexstring to be signed. '010001f0' + {length of salt+message in hex} + salt + message + '0000'
+* Sign the hexstring value once the user confirms
+* Return to dapp
+
+For more details, please see:
+[https://docs.switcheo.network/#signing-messages-for-neo]
+
+Example
+<pre>
+{
+  publicKey: '0241392007396d6ef96159f047967c5a61f1af0871ecf9dc82afbedb68afbb949a',
+  data: '0147fb89d0999e9d8a90edacfa26152fe695ec8b3770dcad522048297ab903822e12472364e254ff2e088fc3ebb641cc24722c563ff679bb1d1623d08bd5863d0d',
+  salt: '058b9e03e7154e4db1e489c99256b7fa',
+  message: 'Hello World!',
+}
+</pre>
+
+====deploy====
+Will deploy a compiled smart contract to the blockchain with the provided input parameters. The GAS cost for deploying the contract will be calculated by the provider, and displayed to the user upon tx acceptance or rejection.
+
+
+'''Method Interface'''
+
+<pre>
+function deploy({
+  name: string,
+  version: string,
+  author: string,
+  email: string,
+  description: string,
+  needsStorage?: boolean,
+  dynamicInvoke?: boolean,
+  isPayable?: boolean,
+  parameterList: string,
+  returnType: string,
+  code: string,
+  network?: string,
+  broadcastOverride?: boolean,
+}): Promise<DeployOutput>
+</pre>
+
+
+'''Input arguments'''
+
+<pre>
+interface DeployArgs {
+  name: string;
+  version: string;
+  author: string;
+  email: string;
+  description: string;
+  needsStorage?: boolean;
+  dynamicInvoke?: boolean;
+  isPayable?: boolean;
+  parameterList: string;
+  returnType: string;
+  code: string;
```

Comment:
`code` should be named `script`, in my opinion.

Reply:
I have no opposition to this. As long as all other wallet providers and dapps are okay with making the update.

Reply:
If people are already using like this, thats fine.